cypress/integration/admin/fileManager/uploadImage.spec.js

```diff
@@ -20,18 +68,288 @@ context("File Manager View - CRUD", () => {
+                    "webiny5_img1"
+                ]
+            },
+            {
+                fileName: "sample_2.jpeg",
+                fileType: "image/jpeg",
+                tagValues: [
+                    "webiny1_img2",
+                    "webiny2_img2",
+                    "webiny3_img2",
+                    "webiny4_img2",
+                    "webiny5_img2"
+                ]
+            },
+            {
+                fileName: "sample.jpeg",
+                fileType: "image/jpeg",
+                tagValues: [
+                    "webiny1_img3",
+                    "webiny2_img3",
+                    "webiny3_img3",
+                    "webiny4_img3",
+                    "webiny5_img3"
+                ]
+            }
+        ];
+
+        cy.fmListFiles({}).then(files => {
+            for (let i = 0; i < files.length; i++) {
+                deleteFile();
+            }
+        });
+
+        // Upload 3 files and add 5 tags per file.
+        fileDetails.forEach(fileDetail => {
+            cy.findByTestId("fm-list-wrapper").dropFile(fileDetail.fileName, fileDetail.fileType);
+            cy.findByText("File upload complete.").should("exist");
+
+            cy.findByTestId("fm-list-wrapper").within(() => {
+                cy.findAllByTestId("fm-list-wrapper-file")
+                    .first()
+                    .within(() => {
+                        cy.findByTestId("fm-file-wrapper-file-info-icon").click({ force: true });
+                    });
+            });
+
+            cy.get("span").contains("File details").should("be.visible");
+
+            cy.get("span").contains("Add tags...").click();
+
+            cy.get('input[placeholder="homepage asset"]').type(fileDetail.tagValues[0]);
+            cy.get("ul > li > span").contains(fileDetail.tagValues[0]).click();
+
+            cy.get('input[placeholder="homepage asset"]').type(fileDetail.tagValues[1]);
+            cy.get("ul > li > span").contains(fileDetail.tagValues[1]).click();
+
+            cy.get('input[placeholder="homepage asset"]').type(fileDetail.tagValues[2]);
+            cy.get("ul > li > span").contains(fileDetail.tagValues[2]).click();
+
+            cy.get('input[placeholder="homepage asset"]').type(fileDetail.tagValues[3]);
+            cy.get("ul > li > span").contains(fileDetail.tagValues[3]).click();
+
+            cy.get('input[placeholder="homepage asset"]').type(fileDetail.tagValues[4]);
+            cy.get("ul > li > span").contains(fileDetail.tagValues[4]).click();
+
+            cy.get("span").contains("Submit").click();
+            cy.contains("Tags successfully updated.").should("be.visible");
+
+            cy.visit("/settings/file-manager/general");
+            cy.findByTestId("apps-menu").click();
+            cy.findByTestId("admin-drawer-footer-menu-file-manager").click();
+        });
+
+        // Assert all the 15 tags are visible.
+        fileDetails.forEach(fileTags => {
```

Comment:
Let's change `fileTags` to `fileDetail` as you had it above. Just a bit easier to read.